clang/lib/CodeGen/CGStmtOpenMP.cpp

```diff
@@ -592,16 +592,22 @@ CodeGenFunction::GenerateOpenMPCapturedStmtFunction(const CapturedStmt &S,
   FunctionArgList Args;
   llvm::MapVector<const Decl *, std::pair<const VarDecl *, Address>> LocalAddrs;
   llvm::DenseMap<const Decl *, std::pair<const Expr *, llvm::Value *>> VLASizes;
+  // AMDGCN does not generate wrapper kernels properly, fails to launch kernel.
+  if (CGM.getTriple().isAMDGCN())
+    NeedWrapperFunction = false;
   SmallString<256> Buffer;
   llvm::raw_svector_ostream Out(Buffer);
   Out << CapturedStmtInfo->getHelperName();
+  bool isKernel = (Out.str().find("__omp_offloading_") != std::string::npos);
```

Comment:
these four lines [buffer, isKernel] are duplicated below, should make a helper function